net/utils.cpp

```diff
@@ -255,43 +257,46 @@ class DefaultResolver : public Resolver {
     ~DefaultResolver() { dnscache_.clear(); }
 
     IPAddr resolve(const char *host) override {
-        auto ctr = [&]() -> IPAddr * {
-            std::vector<IPAddr> addrs;
+        auto ctr = [&]() -> std::list<IPAddr>* {
+            auto addrs = new std::list<IPAddr>();
```

Comment:
consider unordered_set

Reply:
If using unordered_set, get element in a round-robin order would be O(N).

Reply:
OK